jax/_src/scipy/signal.py

```diff
@@ -492,3 +492,110 @@ def welch(x, fs=1.0, window='hann', nperseg=None, noverlap=None, nfft=None,
                    axis=axis, average=average)
 
   return freqs, Pxx.real
+
+
+def _overlap_sum(x, step_size):
+  *batch_shape, segment_len, nframe = x.shape
+  flat_batchsize = np.prod(batch_shape, dtype=np.int64)
+  x = x.reshape((flat_batchsize, segment_len, nframe))
+  out_length = step_size * (nframe - 1) + segment_len
+
+  nstep_per_segment = 1 + (segment_len - 1) // step_size
+  padded_segment_len = nstep_per_segment * step_size
+  pad_x = jnp.pad(x, ((0, 0), (0, padded_segment_len - segment_len), (0, 0)))
+  y = jnp.zeros((flat_batchsize, out_length), dtype=x.dtype)
+  for i in range(nstep_per_segment):
+    shifted = pad_x[:, step_size * i:step_size * (i + 1), :]
+    shifted = shifted.transpose((0, 2, 1)).reshape((flat_batchsize, -1))
+
+    leftpad = step_size * i
+    rightpad = max(0, segment_len - step_size)
+    shifted = jnp.pad(shifted, ((0, 0), (leftpad, rightpad)))[:, :out_length]
+
+    y += shifted
+  return y.reshape(tuple(batch_shape) + (-1,))
+
+
+@_wraps(osp_signal.istft)
+def istft(Zxx, fs=1.0, window='hann', nperseg=None, noverlap=None, nfft=None,
+          input_onesided=True, boundary=True, time_axis=-1, freq_axis=-2):
+  # Input validation
+  if Zxx.ndim < 2:
+    raise ValueError('Input stft must be at least 2d!')
+  freq_axis = canonicalize_axis(freq_axis, Zxx.ndim)
+  time_axis = canonicalize_axis(time_axis, Zxx.ndim)
+  if freq_axis == time_axis:
+    raise ValueError('Must specify differing time and frequency axes!')
+
+  Zxx = jnp.asarray(Zxx, dtype=jax.dtypes.canonicalize_dtype(
+      np.result_type(Zxx, np.complex64)))
```

Comment:
Probably not for this PR, but this kind of thing is used frequently enough that we should probably define a `_promote_dtypes_complex` helper similar to our current `_promote_dtypes_inexact`.

Reply:
I agree, such utility function should be useful.